engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/SelectGameScreen.java

```diff
@@ -169,8 +170,9 @@ private void loadGame(GameInfo item) {
             final GameManifest manifest = item.getManifest();
             config.getWorldGeneration().setDefaultSeed(manifest.getSeed());
             config.getWorldGeneration().setWorldTitle(manifest.getTitle());
-            CoreRegistry.get(GameEngine.class).changeState(new StateLoading(manifest, (isLoadingAsServer()) ?
-                    NetworkMode.DEDICATED_SERVER : NetworkMode.NONE));
+            Optional.ofNullable(CoreRegistry.get(GameEngine.class))
+                    .orElseThrow(() -> new Exception("Failed to get game engine from CoreRegistry"))
```

Comment:
Throwing a generic `Exception` is usually no good idea. We should probably introduce custom Terasology exceptions for cases such like this, but for a start you could throw [IllegalStateException](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalStateException.html) instead.